python/paddle/fluid/parallel_executor.py

```diff
@@ -123,6 +124,10 @@ def __init__(self,
             exec_strategy=exec_strategy,
             share_vars_from=share_vars_from._compiled_program
             if share_vars_from else None)
+
+        if num_trainers > 1:
+            self._compiled_program._build_strategy.is_distribution = True
```

Comment:
May not be true, nccl2 mode should set this to false.

Reply:
Use it for DGC now. And I will fix it in next pr.